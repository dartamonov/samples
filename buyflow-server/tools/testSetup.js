// Tests are placed alongside files under test.
// This file does the following:
// 1. Sets the environment to 'production' so that
//    dev-specific babel config in .babelrc doesn't run.
// 2. Disables Webpack-specific features that Mocha doesn't understand.
// 3. Registers babel for transpiling our code for testing.

// This assures the .babelrc dev config (which includes
// hot module reloading code) doesn't apply for tests.
// process.env.NODE_ENV = 'production';
// process.env.NODE_ENV = 'test';

// Register babel so that it will transpile ES6 to ES5
// before our tests run.
require('babel-register')();
//require("babel-polyfill");
require("babel-jest");
