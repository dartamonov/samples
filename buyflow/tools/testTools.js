import renderer from 'react-test-renderer';

export const takeSnapshot = x => renderer.create(x).toJSON();