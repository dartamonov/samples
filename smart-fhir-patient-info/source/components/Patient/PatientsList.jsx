import React, { Component } from 'react';

export class PatientsList extends Component {
  render() {

    return (
      <div>
        <h1 tabIndex="-1" className="x-heading1">Patients</h1>
        <p>View patients by ID:</p>
        <ul>
          <li><a href="/patient/2744010">2744010</a></li>
          <li><a href="/patient/1316024">1316024</a></li>
        </ul>
      </div>
    );
  }
}

export default PatientsList;
