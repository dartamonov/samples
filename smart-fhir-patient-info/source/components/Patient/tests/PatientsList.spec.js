import React from 'react';
import {takeSnapshot} from '../../../tools/testTools';
import PatientsList from '../PatientsList.jsx';

describe("PatientsList component", () => {
  it('handles patients list', () => {
    const snapshot = takeSnapshot(
      <PatientsList/>
    );

    expect(snapshot).toMatchSnapshot();
  });
});
