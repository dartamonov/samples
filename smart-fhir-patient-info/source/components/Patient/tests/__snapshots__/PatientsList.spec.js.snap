// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientsList component handles patients list 1`] = `
<div>
  <h1
    className="x-heading1"
    tabIndex="-1"
  >
    Patients
  </h1>
  <p>
    View patients by ID:
  </p>
  <ul>
    <li>
      <a
        href="/patient/2744010"
      >
        2744010
      </a>
    </li>
    <li>
      <a
        href="/patient/1316024"
      >
        1316024
      </a>
    </li>
  </ul>
</div>
`;
